starknet/src/tests/proposal_validation_strategies/proposing_power.cairo

```diff
@@ -0,0 +1,236 @@
+#[cfg(test)]
+mod tests {
+    use starknet::syscalls::deploy_syscall;
+    use traits::{TryInto};
+    use starknet::SyscallResult;
+    use array::{ArrayTrait, SpanTrait};
+    use result::ResultTrait;
+    use option::OptionTrait;
+    use sx::voting_strategies::vanilla::{VanillaVotingStrategy};
+    use sx::voting_strategies::merkle_whitelist::{MerkleWhitelistVotingStrategy};
+    use sx::utils::merkle::Leaf;
+    use sx::proposal_validation_strategies::proposing_power::{
+        ProposingPowerProposalValidationStrategy
+    };
+    use sx::interfaces::{
+        IProposalValidationStrategy, IProposalValidationStrategyDispatcher,
+        IProposalValidationStrategyDispatcherTrait
+    };
+    use sx::types::{IndexedStrategy, Strategy, UserAddress};
+    use serde::Serde;
+    use starknet::contract_address_const;
+    use clone::Clone;
+    use sx::tests::test_merkle_whitelist::merkle_utils::{
+        generate_merkle_data, generate_merkle_root, generate_proof
+    };
+
+    // #[test]
+    // #[available_gas(10000000000)]
+    fn test_vanilla_works() {
+        // deploy vanilla voting strategy
+        let (vanilla_contract, _) = deploy_syscall(
+            VanillaVotingStrategy::TEST_CLASS_HASH.try_into().unwrap(), 0, array![].span(), false
+        )
+            .unwrap();
+
+        let vanilla_strategy = Strategy { address: vanilla_contract, params: array![],  };
+
+        // create a proposal validation strategy
+        let (proposal_validation_contract, _) = deploy_syscall(
+            ProposingPowerProposalValidationStrategy::TEST_CLASS_HASH.try_into().unwrap(),
+            0,
+            array![].span(),
+            false
+        )
+            .unwrap();
+
+        let allowed_strategies = array![vanilla_strategy.clone()];
+        let proposal_threshold = 1_u256;
+        let mut params: Array<felt252> = array![];
```

Comment:
cant infer the type here?

Reply:
8d28ee54bc422a0e512b6fba506907962cf8976c